scripts/pylib/power-twister-harness/test_power.py

```diff
@@ -0,0 +1,96 @@
+# Copyright: (c)  2024, Intel Corporation
+
+import json
+import logging
+
+import pytest
+import utils.UtilityFunctions as UtilityFunctions
+from abstract.PowerMonitor import PowerMonitor
+from twister_harness import DeviceAdapter
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize("probe_class", ['stm_powershield'], indirect=True)
+def test_power_harness(probe_class: PowerMonitor, probe_path, request, dut: DeviceAdapter):
+    """
+    This test measures and validates the RMS current values from the power monitor
+    and compares them against expected RMS values.
+
+    Arguments:
+    probe_class -- The Abstract class of the power monitor.
+    probe_path -- The path to the probe device.
+    request -- Request object that provides access to test configuration.
+    dut -- The Device Under Test (DUT) that the power monitor is measuring.
+    """
+
+    # Get test data from the configuration and parse it into a dictionary
+    measurements = request.config.getoption("--testdata")
+    measurements = measurements.replace("'", '"')  # Ensure the data is properly formatted as JSON
+    measurements_dict = json.loads(measurements)
+
+    # Extract individual measurement parameters from the config
+    elements_to_trim = measurements_dict['elements_to_trim']
+    min_peak_distance = measurements_dict['min_peak_distance']
+    min_peak_height = measurements_dict['min_peak_height']
+    peak_padding = measurements_dict['peak_padding']
+    measurement_duration = measurements_dict['measurement_duration']
+    num_of_transitions = measurements_dict['num_of_transitions']
+    expected_rms_values = measurements_dict['expected_rms_values']
+    tolerance_percentage = measurements_dict['tolerance_percentage']
```

Comment:
This can also be a fixture returning a measurement_dict. Providing a test data as a fixture and not as a part of a test is preferable design: the test function will be even cleaner and test data related part will be encapsulated. Data validation would then also be a part of data preparation fixture and if there are issues pytest will report that there was an error during test setup.